crates/polars-time/src/windows/group_by.rs

```diff
@@ -298,42 +310,47 @@ pub(crate) fn group_by_values_iter_window_behind_t(
         TimeUnit::Milliseconds => Duration::add_ms,
     };
 
-    let mut lagging_offset = 0;
-    time.iter().enumerate().map(move |(i, lower)| {
+    let t0 = time[0];
```

Comment:
looks like this is causing the test failure
```
index out of bounds: the len is 0 but the index is 0
```

Reply:
I got other local hypothesis failures that seem unrelated. not familiar with hypohtesis. Is there a way I can get the test out? ^^

Reply:
Ok, got something working. :+1:

Reply:
wait, it's not unrelated, will post something now

Reply:
```python
df = pl.from_repr(
    """
shape: (3, 2)
┌─────────────────────┬───────┐
│ ts                  ┆ value │
│ ---                 ┆ ---   │
│ datetime[ns]        ┆ i64   │
╞═════════════════════╪═══════╡
│ 2000-01-01 00:00:00 ┆ 0     │
│ 2148-01-01 00:00:00 ┆ 0     │
│ 2148-01-02 00:00:00 ┆ 0     │
└─────────────────────┴───────┘"""
).sort('ts')
print(df.group_by_rolling("ts", period='1us', offset='-1d', closed='left').agg(pl.col("value")))
```